lib/matplotlib/ticker.py

```diff
@@ -390,7 +391,36 @@ def __call__(self, x, pos=None):
 
         *x* and *pos* are passed through as-is.
         """
-        return self.func(x, pos)
+        if self._nargs == 1:
+            return self._func(x)
+        return self._func(x, pos)
+
+    @property
+    def func(self):
+        return self._func
+
+    @func.setter
+    def func(self, func):
+        try:
+            sig = inspect.signature(func)
+        except ValueError:
+            self._nargs = 2
+            cbook._warn_external("FuncFormatter may not support "
+                                  f"{func.__name__}. Please look at the "
+                                  "other formatters in `matplotlib.ticker`.")
+        else:
+            try:
+                sig.bind(None, None)
+                self._nargs = 2
+            except TypeError:
+                try:
+                    sig.bind(None)
+                    self._nargs = 1
+                except TypeError:
+                    raise TypeError(f"{func.__name__} must take "
+                                     "at most 2 arguments: "
+                                     "x (required), pos (optional).")
```

Comment:
Maybe I'm over-simplifying, but can we just try?

```suggestion
        self._nargs = 2
        # check whether func really suppors two args
        try:
            func(0, 0)
        except TypeError as e:
            # Only catch the specific error that a single-arg func would raise:
            #     TypeError: func() takes 1 positional argument but 2 were given
            # every other error is not our business here and will raise on
            # actual usage. Since the TypeError will be thrown before any
            # internal errors in the function, this approach is a safe way to
            # identify single-arg functions without introspection need.
            if "1 positional argument" in str(e):
                self._nargs = 1
```

The only risk I see is that python changes its exception message, but i) by testing only against "1 positional argument" were more resilent wrt minor changes, and ii) we should add a test that monitors the exception message thrown in such cases.